### PR TITLE
Create my own enumerable

### PR DESCRIPTION
**Pull request by MattGomb**
Dear Reviewer!

image

In this project, we created a class and a module, as per the requirements. The module mimics the Enumerable mixin of Ruby, and we use it on a short array in our class. Class and module are in separate files.

The solution is verified by the test lines.

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require_relative('my_enumerable')
+class MyList
+  include MyEnumerable
+  def initialize(list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each { |a| block.call a }
+  end
```
Comment (mikethreels) [inline my_list.rb]: - [x] when creating the `each` method please avoid using the already existing method.

File: my_list.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+end
+
+list = MyList.new([1, 2, 3, 4])
```
Comment (mikethreels) [inline my_list.rb]: - [x] the user should be able to add a list of numbers without it being an array already like the below example:
`list = MyList.new(1, 2, 3, 4)`
when running this code in the current application the user receives an error.
you can fix this in the initialize method by doing something similar to below:
```
    def initialize(*args)
        @list = []
        args.each { |arg| @list << arg }
    end
```

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 2 above)
Comment (MattGomb) [reply my_list.rb]: The array issue is fixed, thanks!

File: my_list.rb (hunk 1 above)
Comment (MattGomb) [reply my_list.rb]: Changed to yield

**Review by rustam-tolipov — APPROVED** [review]
Hi Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯

## Optional suggestions
Kindly remove Microverse logo from your Readme file :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**